servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/MutualSslTest.java

```diff
@@ -44,32 +54,58 @@
         this.serverSslProvider = serverSslProvider;
         this.clientSslProvider = clientSslProvider;
+        this.serverListenOptions = serverListenOptions;
+        this.clientOptions = clientOptions;
     }
 
-    @Parameterized.Parameters(name = "server={0} client={1}")
+    @Parameterized.Parameters(name = "server={0} client={1} server opts={2} client opts={3}")
     public static Collection<Object[]> sslProviders() {
         return asList(
-                new Object[]{JDK, JDK},
-                new Object[]{JDK, OPENSSL},
-                new Object[]{OPENSSL, JDK},
-                new Object[]{OPENSSL, OPENSSL}
+                new Object[]{JDK, JDK, emptyMap(), emptyMap()},
+                new Object[]{JDK, OPENSSL, emptyMap(), emptyMap()},
+                new Object[]{OPENSSL, JDK, emptyMap(), emptyMap()},
+                new Object[]{OPENSSL, OPENSSL, emptyMap(), emptyMap()},
+                new Object[]{JDK, JDK, emptyMap(), clientTcpFastOpenOptions()},
+                new Object[]{JDK, OPENSSL, emptyMap(), clientTcpFastOpenOptions()},
+                new Object[]{OPENSSL, JDK, emptyMap(), clientTcpFastOpenOptions()},
+                new Object[]{OPENSSL, OPENSSL, emptyMap(), clientTcpFastOpenOptions()},
+                new Object[]{JDK, JDK, serverTcpFastOpenOptions(), emptyMap()},
+                new Object[]{JDK, OPENSSL, serverTcpFastOpenOptions(), emptyMap()},
+                new Object[]{OPENSSL, JDK, serverTcpFastOpenOptions(), emptyMap()},
+                new Object[]{OPENSSL, OPENSSL, serverTcpFastOpenOptions(), emptyMap()},
+                new Object[]{JDK, JDK, serverTcpFastOpenOptions(), clientTcpFastOpenOptions()},
+                new Object[]{JDK, OPENSSL, serverTcpFastOpenOptions(), clientTcpFastOpenOptions()},
+                new Object[]{OPENSSL, JDK, serverTcpFastOpenOptions(), clientTcpFastOpenOptions()},
+                new Object[]{OPENSSL, OPENSSL, serverTcpFastOpenOptions(), clientTcpFastOpenOptions()}
         );
     }
 
     @Test
     public void mutualSsl() throws Exception {
-        try (ServerContext serverContext = HttpServers.forAddress(localAddress(0))
+        HttpServerBuilder serverBuilder = HttpServers.forAddress(localAddress(0))
                 .secure()
                 .provider(serverSslProvider)
                 .clientAuth(REQUIRE)
                 .trustManager(DefaultTestCerts::loadClientCAPem)
-                .commit(DefaultTestCerts::loadServerPem, DefaultTestCerts::loadServerKey)
-                .listenBlockingAndAwait((ctx, request, responseFactory) -> responseFactory.ok());
-             BlockingHttpClient client = HttpClients.forSingleAddress(serverHostAndPort(serverContext))
+                .commit(DefaultTestCerts::loadServerPem, DefaultTestCerts::loadServerKey);
+        for (@SuppressWarnings("rawtypes") Entry<SocketOption, Object> entry : serverListenOptions.entrySet()) {
```

Comment:
SocketOption<?> perhaps? 
Sadly you can't introduce a type variable here to say <T> Entry<SocketOption<T>, ? extends T>

Reply:
Then the generic types don't line up for `builder.socketOption(option, entry.getValue());` which causes issues.

> reason: no instance(s) of type variable(s) exist so that Object conforms to capture of ? inference variable T has incompatible bounds: equality constraints: capture of ? lower bounds: Object